addons/controls/boolean.js

```diff
@@ -0,0 +1,43 @@
+const boolean = (name, storyObj, config, preferredValue) => {
+	if (typeof config === 'string' || config == null) {
+		// Config wasn't set, or was omitted, causing the preferredValue to be the last value. Reassignment dipsy-doodle.
+		preferredValue = config;
+		config = {};
+	}
+
+	// If there is no `args` object on the storyObj object
+	if (!storyObj.args) {
+		storyObj.args = {};
+	}
+
+	// If there is no `argTypes` object on the storyObj object
+	if (!storyObj.argTypes) {
+		storyObj.argTypes = {};
+	}
+
+	// If there is no `defaultProps` object on the config object
+	if (!config.defaultProps) {
+		config.defaultProps = {};
+	}
+
+	// If there's no group ID but there is a display name, use that for the group ID
+	if (config.displayName && !config.groupId) {
+		config.groupId = config.displayName;
+	}
+
+	// Set false for default boolean props that are not defined.
+	const defaultValue = config.defaultProps[name] != null ? config.defaultProps[name] : false;
+
+	storyObj.args[name] = preferredValue != null ? preferredValue : defaultValue;
```

Comment:
Have you considered `nullify` before you removed it?

Reply:
I've checked by testing without it. If the value is `null` or `''` or `false`, it is automatically nullified because the addon-controls basically infer the type of value. (referred to `https://storybook.js.org/docs/react/essentials/controls#fully-custom-args`)
So I think `nullify` is not needed anymore.

Reply:
Please make sure empty values can be selected from controls.

Reply:
I've added using `nullify` as before and changed if the selected value is falsy, it shows `undefined` text so the value can be selected in the controls.